src/Microsoft.DotNet.Interactive.HttpRequest.Tests/ParserTests.cs

```diff
@@ -0,0 +1,595 @@
+                Authorization: token xxx
+
+                <request>
+                    <name>sample</name>
+                    <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+                </request>
+                """);
+
+            using var _ = new AssertionScope();
+
+            var requestNode = result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+            requestNode.HeadersNode.HeaderNodes.Count.Should().Be(2);            
+            requestNode.BodyNode.Text.Should().Be(
+                """
+                <request>
+                    <name>sample</name>
+                    <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+                </request>
+                """);
+        }
+
+        [Fact]
+        public void header_separator_is_present()
+        {
+            var result = HttpRequestParser.Parse(
+                """
+                POST https://example.com/comments HTTP/1.1
+                Content-Type: application                                                                                                            
+                """);         
+
+            var requestNode = result.SyntaxTree.RootNode
+                    .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+            requestNode.HeadersNode.ChildNodes.Should().ContainSingle<HttpHeaderNode>().Which
+                .ChildNodes.Should().ContainSingle<HttpHeaderSeparatorNode>();
+        }
+
+        [Fact]
+        public void body_separator_is_present()
+        {
+            var result = HttpRequestParser.Parse(
+            """
+            POST https://example.com/comments HTTP/1.1
+            Content-Type: application/xml
+            Authorization: token xxx
+
+            <request>
+                <name>sample</name>
+                <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+            </request>
+            """);         
+
+            var requestNode = result.SyntaxTree.RootNode
+                    .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+            requestNode.BodySeparatorNode.TextWithTrivia.Should().Be("\r\n");          
+        }
+
+        [Fact]
+        public void body_without_header_header_is_parsed_correctly()
+        {
+            var result = HttpRequestParser.Parse(
+                """
+                POST https://example.com/comments HTTP/1.1
+
+                <request>
+                    <name>sample</name>
+                    <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+                </request>
+                """);
+
+            using var _ = new AssertionScope();
+
+            var requestNode = result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+            requestNode.HeadersNode.HeaderNodes.Count.Should().Be(0);
+        }
+
+        //TODO: Is it an error to include a body with no headers?
```

Comment:
Let's remove this TODO - I think this will bubble naturally if it is required when we implement all the error reporting.